components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -742,10 +776,11 @@ void AIChatTabHelper::OnAPIStreamDataComplete(
   }
 
   if (!success) {
-    // TODO(petemill): show error state separate from assistant message
-    AddToConversationHistory(ConversationTurn{
-        CharacterType::ASSISTANT, ConversationTurnVisibility::VISIBLE,
-        l10n_util::GetStringUTF8(IDS_CHAT_UI_API_ERROR)});
+    if (net::HTTP_TOO_MANY_REQUESTS == result.response_code()) {
+      SetAPIError(mojom::APIError::RateLimitReached);
```

Comment:
When does this get cleared? I would think when any new request is started it would clear the rate limit error. I know we've disabled that now based on any error but I think the rate limit experience might need to be different as there's no way to retry for this conversation. We can do that in a follow-up.

Reply:
I agree: https://github.com/brave/brave-browser/issues/32602